relay/polkadot/src/lib.rs

```diff
@@ -621,6 +622,70 @@ impl pallet_election_provider_multi_phase::Config for Runtime {
 	type ElectionBounds = ElectionBounds;
 }
 
+/// Parameters that can be adjusted at runtime.
+#[dynamic_params(RuntimeParameters, pallet_parameters::Parameters::<Runtime>)]
+pub mod dynamic_params {
+	use super::*;
+
+	/// Parameters used to calculate inflation.
+	#[dynamic_pallet_params]
+	#[codec(index = 0)]
+	pub mod inflation {
+		/// The fixed amount minted per year.
+		/// Default set to 8% of 15_011_657_390_566_252_333, following execution of [Referendum 1139](https://polkadot.subsquare.io/referenda/1139), block hash: `0x39422610299a75ef69860417f4d0e1d94e77699f45005645ffc5e8e619950f9f`.
+		#[codec(index = 0)]
+		pub static YearlyEmission: u128 = 1_200_932_591_245_300_186;
+
+		/// The percentage of inflation diverted to the treasury.
+		/// Default to 15%, as per [Referendum 1139](https://polkadot.subsquare.io/referenda/1139).
+		#[codec(index = 1)]
+		pub static PercentToTreasury: Perquintill = Perquintill::from_percent(15);
+	}
+}
+
+#[cfg(feature = "runtime-benchmarks")]
+impl Default for RuntimeParameters {
+	fn default() -> Self {
+		RuntimeParameters::Inflation(dynamic_params::inflation::Parameters::YearlyEmission(
+			dynamic_params::inflation::YearlyEmission,
+			Some(1_200_932_591_245_300_186u128),
+		))
+	}
+}
+
+/// Defines which origins can modify which dynamic parameters.
+pub struct DynamicParametersOrigins;
+impl frame_support::traits::EnsureOriginWithArg<RuntimeOrigin, RuntimeParametersKey>
+	for DynamicParametersOrigins
+{
+	type Success = ();
+
+	fn try_origin(
+		origin: RuntimeOrigin,
+		key: &RuntimeParametersKey,
+	) -> Result<Self::Success, RuntimeOrigin> {
+		use crate::RuntimeParametersKey::*;
+
+		match key {
+			Inflation(_) => frame_system::ensure_root(origin.clone()),
```

Comment:
This pattern makes sense when there are many parameters, but for this simple case you can just do `frame_system::ensure_root(origin)`.